mlir/unittests/translation/test_translation.cpp

```diff
@@ -188,541 +269,653 @@ TEST_F(ImportTest, Reset0) {
-  const auto* checkString = "CHECK: mqtref.v()";
-
-  ASSERT_TRUE(checkOutput(checkString, outputString));
-}
-
-TEST_F(ImportTest, Vdg) {
-  qc::QuantumComputation qc(1);
-  qc.vdg(0);
-
-  auto module = translateQuantumComputationToMLIR(context.get(), qc);
-
-  const auto outputString = getOutputString(&module);
-  const auto* checkString = "CHECK: mqtref.vdg()";
-
-  ASSERT_TRUE(checkOutput(checkString, outputString));
-}
-
-TEST_F(ImportTest, U) {
-  qc::QuantumComputation qc(1);
-  qc.u(0.1, 0.2, 0.3, 0);
-
-  auto module = translateQuantumComputationToMLIR(context.get(), qc);
-
-  const auto outputString = getOutputString(&module);
-  const auto* checkString =
-      "CHECK: mqtref.u( static [1.000000e-01, 2.000000e-01, 3.000000e-01])";
-
-  ASSERT_TRUE(checkOutput(checkString, outputString));
+  return result;
 }
 
-TEST_F(ImportTest, U2) {
-  qc::QuantumComputation qc(1);
-  qc.u2(0.1, 0.2, 0);
+class OperationTestIfRegister
+    : public ImportTest,
+      public ::testing::WithParamInterface<TestCaseIfRegister> {};
 
-  auto module = translateQuantumComputationToMLIR(context.get(), qc);
-
-  const auto outputString = getOutputString(&module);
-  const auto* checkString =
-      "CHECK: mqtref.u2( static [1.000000e-01, 2.000000e-01])";
-
-  ASSERT_TRUE(checkOutput(checkString, outputString));
-}
-
-TEST_F(ImportTest, P) {
-  qc::QuantumComputation qc(1);
-  qc.p(0.1, 0);
-
-  auto module = translateQuantumComputationToMLIR(context.get(), qc);
-
-  const auto outputString = getOutputString(&module);
-  const auto* checkString = "CHECK: mqtref.p( static [1.000000e-01])";
-
-  ASSERT_TRUE(checkOutput(checkString, outputString));
-}
-
-TEST_F(ImportTest, SX) {
-  qc::QuantumComputation qc(1);
-  qc.sx(0);
-
-  auto module = translateQuantumComputationToMLIR(context.get(), qc);
+TEST_P(OperationTestIfRegister, EmitsExpectedOperation) {
+  const auto& param = GetParam();
 
-  const auto outputString = getOutputString(&module);
-  const auto* checkString = "CHECK: mqtref.sx()";
-
-  ASSERT_TRUE(checkOutput(checkString, outputString));
-}
-
-TEST_F(ImportTest, SXdg) {
-  qc::QuantumComputation qc(1);
-  qc.sxdg(0);
+  QuantumComputation qc(1);
+  const auto& creg = qc.addClassicalRegister(1);
+  qc.measure(0, 0);
+  qc.if_(X, 0, creg, 1U, param.comparisonKind);
 
   auto module = translateQuantumComputationToMLIR(context.get(), qc);
 
-  const auto outputString = getOutputString(&module);
-  const auto* checkString = "CHECK: mqtref.sxdg()";
-
-  ASSERT_TRUE(checkOutput(checkString, outputString));
-}
-
-TEST_F(ImportTest, Rx) {
-  qc::QuantumComputation qc(1);
-  qc.rx(0.1, 0);
-
-  auto module = translateQuantumComputationToMLIR(context.get(), qc);
+  // Run passes
+  mlir::PassManager passManager(context.get());
+  passManager.addPass(mlir::createMem2Reg());
+  passManager.addPass(mlir::createRemoveDeadValuesPass());
+  passManager.addPass(mlir::createCanonicalizerPass());
+  passManager.addPass(mlir::createMem2Reg());
```

Comment:
I am pretty sure one of these `mem2reg` passes is redundant. I just can't remember which. I also kind of liked that this was moved to a function and that all IfElse tests simply called that function.